site-scrapers/Hannaford.js

```diff
@@ -63,10 +65,77 @@ async function ScrapeWebsiteData(browser) {
                 const filename =
                     "hannaford-zip-" + loc.zip + "-date-" + today + ".png";
                 await page.screenshot({ path: filename });
+                await saveBodyHtml(page);
             }
             results[loc.zip] = result;
         }
     }
 
     return results;
 }
+
+async function writeFile(filename, data) {
+    return new Promise((resolve, reject) => {
+        fs.writeFile(filename, data, (err) => {
+            if (err) {
+                reject(err);
+            } else {
+                resolve();
+            }
+        });
+    });
+}
+
+async function saveBodyHtml(page) {
+    // It appears puppeteer does not a generic page.waitForNetworkIdle
+    // so for now we just assume all XHR requests will complete within
+    // a reasonable timeframe, for now 10s.
```

Comment:
this is one second

Reply:
Indeed, I was going to change that to 10 once I knew it worked.